packages/sdks/web-component/src/lib/descope-wc/DescopeWc.ts

```diff
@@ -157,31 +161,62 @@ class DescopeWc extends BaseDescopeWc {
       }
     | undefined;
 
-  async loadSdkScripts() {
-    const flowConfig = await this.getFlowConfig();
+  loadSdkScripts(flowConfig: FlowConfig) {
     const scripts = flowConfig.sdkScripts;
+    if (!scripts?.length) {
+      return null;
+    }
 
-    scripts?.forEach(async (script) => {
-      const module = await loadSdkScript(script.id);
-      module(
-        script.initArgs as any,
-        { baseUrl: this.baseUrl },
-        (result: string) => {
-          // update the context with the result, under the `resultKey` key
-          this.dispatchEvent(
-            new CustomEvent('components-context', {
-              detail: {
-                // we store the result with script.id prefix to avoid conflicts with other scripts results
-                // that may have the same key
-                [getScriptResultPath(script.id, script.resultKey)]: result,
-              },
-              bubbles: true,
-              composed: true,
-            }),
-          );
+    const createScriptCallback =
+      (
+        script: {
+          id: string;
+          resultKey?: string;
         },
-      );
+        resolve: (value: any) => void,
+      ) =>
+      (result: string) => {
+        this.dispatchEvent(
+          // update the context with the result, under the `resultKey` key
+          new CustomEvent('components-context', {
+            detail: {
+              // we store the result with script.id prefix to avoid conflicts with other scripts results
+              // that may have the same key
+              [getScriptResultPath(script.id, script.resultKey)]: result,
+            },
+            bubbles: true,
+            composed: true,
+          }),
+        );
+        resolve(script.id);
+      };
+
+    const promises = Promise.all(
+      scripts?.map(async (script) => {
+        const module = await loadSdkScript(script.id);
+        return new Promise((resolve, reject) => {
+          try {
+            const unloadScript = module(
+              script.initArgs as any,
+              { baseUrl: this.baseUrl },
+              createScriptCallback(script, resolve),
+            );
+            this.nextRequestStatus.subscribe(() => {
+              this.loggerWrapper.debug('Unloading script', script.id);
+              unloadScript?.();
```

Comment:
@orius123 in this case unloadScript will unload the module and each next will have to load the module again if requested, right?